quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java

```diff
@@ -268,9 +278,79 @@ SolverConfigBuildItem recordAndRegisterBuildTimeBeans(CombinedIndexBuildItem com
         return new SolverConfigBuildItem(allSolverConfig, generatedGizmoClasses);
     }
 
-    private SolverConfig generateSolverConfig(ClassLoader classLoader, IndexView indexView,
-            BuildProducer<ReflectiveHierarchyBuildItem> reflectiveHierarchyClass, String solverName,
-            Set<Class<?>> reflectiveClassSet) {
+    private void assertSolverConfigSolutionClasses(IndexView indexView, Map<String, SolverConfig> allSolverConfig) {
+        // Validate the solution class        
+        Collection<AnnotationInstance> annotationInstances = indexView.getAnnotations(DotNames.PLANNING_SOLUTION);
+        // No solution class
+        if (annotationInstances.isEmpty()) {
+            throw new IllegalStateException(
+                    "No classes found with a @%s annotation.".formatted(PlanningSolution.class.getSimpleName()));
+        }
+        // Multiple classes and single solver
+        if (annotationInstances.size() > 1 && allSolverConfig.size() == 1) {
+            throw new IllegalStateException("Multiple classes (%s) found with a @%s annotation.".formatted(
+                    convertAnnotationInstancesToString(annotationInstances), PlanningSolution.class.getSimpleName()));
+        }
+        // Multiple classes and at least one solver config does not specify the solution class
+        List<String> emptySolutionClasses = allSolverConfig.entrySet().stream()
+                .filter(e -> e.getValue().getSolutionClass() == null)
+                .map(Map.Entry::getKey)
+                .toList();
+        if (annotationInstances.size() > 1 && !emptySolutionClasses.isEmpty()) {
+            throw new IllegalStateException(
+                    """
+                            Multiple classes (%s) found with a @%s annotation and some solver configs ([%s]) does not specify it.
```

Comment:
I'd rephrase the first sentence. "Some solver configs (...) don't specify a @PlanningSolution class, yet there are multiple available (...) on the classpath."